caddyconfig/httpcaddyfile/tlsapp.go

```diff
@@ -295,14 +291,48 @@ func (st ServerType) buildTLSApp(
 		tlsApp.Automation.Policies = append(tlsApp.Automation.Policies, internalAP)
 	}
 
+	// if there are any global options set for issuers (ACME ones in particular), make sure they
+	// take effect in every automation policy that does not have any issuers
+	if tlsApp.Automation != nil {
+		globalEmail := options["email"]
+		globalACMECA := options["acme_ca"]
+		globalACMECARoot := options["acme_ca_root"]
+		globalACMEDNS := options["acme_dns"]
+		globalACMEEAB := options["acme_eab"]
+		hasGlobalACMEDefaults := globalEmail != nil || globalACMECA != nil || globalACMECARoot != nil || globalACMEDNS != nil || globalACMEEAB != nil
+		if hasGlobalACMEDefaults {
+			for _, ap := range tlsApp.Automation.Policies {
+				if len(ap.Issuers) == 0 {
+					acme, zerosslACME := new(caddytls.ACMEIssuer), new(caddytls.ACMEIssuer)
+					zerossl := &caddytls.ZeroSSLIssuer{ACMEIssuer: zerosslACME}
+					ap.Issuers = []certmagic.Issuer{acme, zerossl} // TODO: keep this in sync with Caddy's other issuer defaults elsewhere, like in caddytls/automation.go (DefaultIssuers).
```

Comment:
This will be tried in order, right? Unlike ACME challenges which are random

Reply:
https://github.com/caddyserver/caddy/pull/3862/files?diff=split&w=1#diff-29717d7c793dbc1d844ba62b3f2cd39faaa2c97356d9ea51da21af99571cf54aR241 answered my question

Reply:
Yes, although I don't want to make that guarantee. We may end up "load balancing" issuers someday. So, make no assumptions which one is first.

Reply:
Right - I just wanted to confirm how it'll behave for now 👍